internal/peermanager/peerwantmanager.go

```diff
@@ -36,20 +43,24 @@ type peerWant struct {
 // number of active want-blocks (ie sent but no response received)
 func newPeerWantManager(wantBlockGauge Gauge) *peerWantManager {
 	return &peerWantManager{
+		broadcastWants: cid.NewSet(),
 		peerWants:      make(map[peer.ID]*peerWant),
 		wantPeers:      make(map[cid.Cid]map[peer.ID]struct{}),
 		wantBlockGauge: wantBlockGauge,
 	}
 }
 
-// AddPeer adds a peer whose wants we need to keep track of
-func (pwm *peerWantManager) addPeer(p peer.ID) {
+// addPeer adds a peer whose wants we need to keep track of. It returns the
+// current list of broadcast wants that should be sent to the peer.
+func (pwm *peerWantManager) addPeer(p peer.ID) []cid.Cid {
```

Comment:
Too overloaded? We could also add a second function to get the current broadcast wants.